doc/neps/nep-0054-simd-cpp-highway.rst

```diff
@@ -0,0 +1,369 @@
+Intrinsics as described in `NEP 38`_. This is a complex and multi-faceted
+decision - this NEP is an attempt to describe the trade-offs involved and
+what would need to be done.
+
+Motivation and Scope
+--------------------
+
+We want to refactor the C-based Universal Intrinsics (see :ref:`NEP 38
+<NEP38>`) to C++. Along the way, the `Google Highway`_ devs proposed using
+Highway rather than our Universal Intrinsics.
+    
+The move from C to C++ is motivated by (a) code readability and ease of
+development, (b) the need to add support for sizeless SIMD instructions (e.g.,
+ARM's SVE, RISC-V's RVV).
+
+As an example of the readability improvement, here is a typical line of C code
+from our current C universal intrinsics framework:
+
+.. code::
+
+   // The @name@ is the numpy-specific templating in .c.src files
+   npyv_@sfx@  a5 = npyv_load_@sfx@(src1 + npyv_nlanes_@sfx@ * 4);
+
+This will change (as implemented in PR `gh-21057`_) to:
+
+.. code:: C++
+
+   auto a5 = Load(src1 + nlanes * 4);
+
+If the above C++ code were to use Highway under the hood it would look quite
+similar, it uses similarly understandable names as ``Load`` for individual
+portable intrinsics.
+
+The ``@sfx`` in the C version above is the template variable for type
+identifiers, e.g.: ``#sfx = u8, s8, u16, s16, u32, s32, u64, s64, f32, f64#``.
+Explicit use of bitsize-encoded types like this won't work for sizeless SIMD
+instruction sets. With C++ this is easier to handle; PR `gh-21057`_ shows how
+and contains more complete examples of what the C++ code will look like.
+
+The scope of this NEP includes discussing most relevant aspects of adopting
+Google Highway to replace our current Universal Intrinsics framework, including
+but not limited to:
+
+- Maintainability, domain expertise availability, ease of onboarding new
+  contributor, and other social aspects,
+- Key technical differences and constraints that may impact NumPy's internal
+  design or performance,
+- Build system related aspects,
+- Release timing related aspects.
+
+Out of scope (at least for now) is revisiting other aspects of our current SIMD
+support strategy:
+
+- accuracy vs. performance trade-offs when adding SIMD support to a function
+- use of SVML and x86-simd-sort (and possibly its equivalents for aarch64)
+- pulling in individual bits or algorithms of Highway (as in `gh-24018`_) or
+  SLEEF (as discussed in that same PR)
+
+
+Usage and Impact
+----------------
+
+N/A - there will be no significant user-visible changes.
+
+
+Backward compatibility
+----------------------
+
+There will be no changes in user-facing Python or C APIs: all the methods to
+control compilation and runtime CPU feature selection should remain, although
+there may be some changes due to moving to C++ without regards to the
+Highway/Universal Intrinsics choice.
+
+The naming of the CPU features in Highway is different from that of the
+Universal Intrinsics. Highway uses clusters of features based on physical CPU
```

Comment:
As far as I know, they're just differently named, such as `HWY_NEON_WITHOUT_AES` and `HWY_NEON` instead of `ASIMD` from NumPy. The features in https://github.com/google/highway/#targets are targeted towards specific x86 CPUs, but I think that's different and only applies to that platform as far as I can see.